forecast/preprocessor.py

```diff
@@ -68,17 +68,26 @@ def get_dataframe(self):
         """
         return self._df
 
-    def get_grouped_dataframe_seconds(self):
+    def get_grouped_dataframe_interval(self, interval):
         """
         Get the pre-grouped version of query log data.
 
+        Parameters
+        ----------
+        interval : pd.TimeDelta
+            time interval to group and count the query templates
+
         Returns
         -------
         grouped_df : pd.DataFrame
             Dataframe containing the pre-grouped query log data.
             Grouped on query template and log time.
         """
-        return self._grouped_df_sec
+        gb = self._df.groupby('query_template').resample(interval).size()
+        grouped_df = pd.DataFrame(gb, columns=['count'])
+        grouped_df.drop('', axis=0, level=0, inplace=True)
+        del gb
```

Comment:
For my education: we need to manually clean up gb??

Reply:
My mistake, originally this was part of a jupyter notebook and I was trying to save memory.